lib/assets/javascripts/turbograft/page.coffee

```diff
@@ -25,3 +25,42 @@ Page.refresh = (options = {}, callback) ->
 
 Page.open = ->
   window.open(arguments...)
+
+# Providing hooks for objects to set up destructors:
+onReplaceCallbacks = []
+
+# e.g., Page.onReplace(node, unbindListenersFnc)
+# unbindListenersFnc will be called if the node in question is partially replaced
+# or if a full replace occurs.  It will be called only once
+Page.onReplace = (node, callback) ->
+  throw new Error("Node and callback must both be specified") if !node || !callback
+  onReplaceCallbacks.push({node, callback})
+
+# from http://davidwalsh.name/check-parent-node
```

Comment:
Maybe the comment should also specify that the else is for old firefox compatibility